src/i18n.ts

```diff
@@ -0,0 +1,75 @@
+import i18n from "i18next";
+import { getRuntime } from "@yext/pages/util";
+import {
+  useTranslation as usei18nextTranslation,
+  initReactI18next,
+} from "react-i18next";
+import LanguageDetector from "i18next-browser-languagedetector";
+import type { TemplateRenderProps } from "src/types/entities";
+
+declare global {
+  interface Window {
+    __INITIAL__DATA__: TemplateRenderProps;
+  }
+}
+
+const detector = {
+  name: "pagesJSDetector",
+
+  lookup() {
+    if (getRuntime().name === "browser") {
+      return window.__INITIAL__DATA__.document.locale as string;
+    }
+  },
+
+  cacheUserLanguage() {
+    // Intentionally unimplemented
+  },
+};
+
+const languageDetector = new LanguageDetector();
+languageDetector.addDetector(detector);
+
+const dev = import.meta.env.DEV;
+const lookupOrder = dev ? ["querystring"] : ["pagesJSDetector"];
+
+i18n
+  .use(initReactI18next) // passes i18n down to react-i18next
+  .use(languageDetector)
+  .init({
+    compatibilityJSON: "v3",
+    returnEmptyString: false,
+    detection: {
+      order: lookupOrder,
+      lookupQuerystring: "locale",
+    },
+
+    interpolation: {
+      escapeValue: false, // react already safes from xss
+    },
+  });
+
+export async function getTranslations(localeCode: string) {
+  return await import(`../locales/${localeCode}/translation.json`)
+    .then((module) => ({ [localeCode]: { translation: module.default } }))
+    .catch((_) => {
+      console.error(`unable to load translations for: ${localeCode}`);
+      return {};
+    });
+}
+
+export function useTranslation() {
```

Comment:
This is a custom wrapper for the `t` translation function. 

The expected use case for i18next seems a little different than what we're used to. They seem to expect that translation files look like this:
```
{
  "greeting": "Hello World!"
}
```

and are used like `<div>Localized Greeting: {t("greeting", "Hi")}</div>` with the first param being a message key and the second param being fallback text. We're used to using the actual text to be translated as the key, so this function makes it so that: `t` automatically uses the key as the fallback text.